### ISSUE TITLE
[Question]: 输出的信息是单卡的吗？单机比2机interval_samples_per_second低

### ISSUE THREAD

**Issue by CX26-CX** [PaddlePaddle/PaddleNLP#7417]
### 请提出你的问题

训练GPT3-13B，使用脚本如下：
```python
task_name="gpt3_hybrid"
# export PYTHONPATH="PaddleNLP/"
export FLAGS_cudnn_deterministic=True
log_dir="log_nfs"
# rm -rf $log_dir

ip_list="×××.×××.××.××,×××.×××.××.××"
#     --ips="${ip_list}" \
python3 -m paddle.distributed.launch \
    --ips="${ip_list}" \
    --gpus "0,1,2,3,4,5,6,7" \
    --log_dir ${log_dir} \
    llm/gpt-3/run_pretrain.py \
    --model_type "gpt" \
    --model_name_or_path "gpt3-13B-en" \
    --tokenizer_name_or_path "gpt3-13B-en" \
    --input_dir "./data" \
    --output_dir "output_nfs/$task_name" \
    --split 949,50,1 \
    --max_seq_length 1024 \
    --per_device_train_batch_size 1 \
    --per_device_eval_batch_size 2 \
    --tensor_parallel_degree 4 \
    --pipeline_parallel_degree 4 \
    --fuse_attention_qkv 0 \
    --use_flash_attention 0 \
    --fp16  \
    --fp16_opt_level "O2"  \
    --scale_loss 1024 \
    --learning_rate 0.00001 \
    --min_learning_rate 0.000005 \
    --max_steps 10000 \
    --save_steps 5000 \
    --weight_decay 0.01 \
    --warmup_ratio 0.01 \
    --max_grad_norm 1.0 \
    --logging_steps 1\
    --dataloader_num_workers 1 \
    --sharding "stage2" \
    --eval_steps 1000 \
    --report_to "visualdl" \
    --disable_tqdm true \
    --recompute 1 \
    --gradient_accumulation_steps 2 \
    --do_train \
    --do_eval \
    --device "gpu" \
```
2机16张卡训练的输出为[2023-11-09 10:06:02,232] [    INFO] - loss: 11.69494629, learning_rate: 4e-07, global_step: 3, interval_runtime: 4.9999, interval_samples_per_second: 0.4000113108972245, interval_steps_per_second: 0.20000565544861226, epoch: 0.0003
单机8卡的输出为：[32m[2023-10-11 09:30:46,361] [    INFO][0m - loss: 10.13138199, learning_rate: 1.1e-06, global_step: 10, interval_runtime: 13.6887, interval_samples_per_second: 0.2922124206637033, interval_steps_per_second: 0.07305310516592582, epoch: 0.001[0m
这个interval_samples_per_second没有明显增长。。看了一下代码，这里算的是单张卡的ips吗？不确定，麻烦问一下这段输出的参数含义

**Comment by CX26-CX** [comment]
有所有参数的介绍吗？这个dataset_world_size什么含义还有dataset_rank

**Comment by ZHUI** [comment]
1. 是全局的速度。
2. 表示有几路独立的数据流。

你速度慢的主要问题是，gradient_accumulation_steps设置太小了。使用pp的话，需要设置为 pp的2-4倍大小为宜

**Comment by CX26-CX** [comment]
修改后运行一段时间会报错是什么原因
![image](https://github.com/PaddlePaddle/PaddleNLP/assets/71750397/180a88bc-0a2e-48a6-9cb4-ef7d8edd7775)

**Comment by CX26-CX** [comment]
现在是python3 -m paddle.distributed.launch \
    --ips="${ip_list}" \
    --gpus "0,1,2,3,4,5,6,7" \
    --log_dir ${log_dir} \
    llm/gpt-3/run_pretrain.py \
    --model_type "gpt" \
    --model_name_or_path "gpt3-13B-en" \
    --tokenizer_name_or_path "gpt3-13B-en" \
    --input_dir "./data" \
    --output_dir "output_nfs/$task_name" \
    --split 949,50,1 \
    --max_seq_length 1024 \
    --per_device_train_batch_size 2 \
    --per_device_eval_batch_size 32 \
    --tensor_parallel_degree 4 \
    --pipeline_parallel_degree 4 \
    --fuse_attention_qkv 0 \
    --use_flash_attention 0 \
    --fp16  \
    --fp16_opt_level "O2"  \
    --scale_loss 1024 \
    --learning_rate 0.00001 \
    --min_learning_rate 0.000005 \
    --max_steps 10000 \
    --save_steps 200 \
    --weight_decay 0.01 \
    --warmup_ratio 0.01 \
    --max_grad_norm 1.0 \
    --logging_steps 1\
    --dataloader_num_workers 1 \
    --sharding "stage2" \
    --eval_steps 1000 \
    --report_to "visualdl" \
    --disable_tqdm true \
    --recompute 1 \
    --gradient_accumulation_steps 16 \
    --do_train \
    --do_eval \
    --device "gpu" \

**Comment by CX26-CX** [comment]
还是2机16卡
